scipy/special/_spfun_stats.py

```diff
@@ -105,3 +105,39 @@ def multigammaln(a, d):
     res = (d * (d-1) * 0.25) * np.log(np.pi)
     res += np.sum(loggam([(a - (j - 1.)/2) for j in range(1, d+1)]), axis=0)
     return res
+
+
+def multivariate_betaln(alpha):
+    r"""Returns the log of the multivariate beta function.
+
+    Parameters
+    ----------
+    alpha : ndarray.
+
+    Returns
+    -------
+    res : float
+        The log of the multivariate beta at the given point `alpha`.
+
+    Notes
+    -----
+    The log of the multivariate beta function is defined as a function of
+    the vector :math:`\mathbf{\alpha}=(\alpha_1,\alpha_2,...,\alpha_n)`:
+
+    .. math::
+
+        \ln B(\mathbf{\alpha}) = \ln\left(\frac{\Gamma(\alpha_1)
+        \Gamma(\alpha_2)...\Gamma(\alpha_n)}{\Gamma(\alpha_1+\alpha_2+
+        ...+\alpha_n)}\right)
+
+    Examples
+    --------
+    Compute the log of the multivariate beta function at one point.
+
+    >>> from scipy.special import multivariate_betaln
+    >>> multivariate_betaln([1, 2, 3])
+    -4.0943445622221
+    """
+
+    alpha = np.asarray(alpha)
+    return np.sum(loggam(alpha)) - loggam(np.sum(alpha))
```

Comment:
This doesn't actually behave like a `gufunc` because it doesn't handle higher dimensional arrays. For example, if you give it an array with shape `(3, 2)` it will return a scalar. Currently on this branch

```python
import numpy as np
import scipy.special as special

alpha = np.array([[3.0, 4.0, 5.0], [1.0, 2.0, 3.0]])
print(special.multivariate_betaln(alpha))
```

results in `-27.148965789441` but I think by default it should reduce along the last axis and the result should be `array([-11.83934737,  -4.09434456])`

Things should just work if you add an axis keyword argument and pass it through to the sums though. Adding `keepdims` and passing it through to the sums would be useful too.

We might want to implement this as a gufunc using the NumPy C API at some point, in which case we'd have to think about keeping the signatures compatible. I think making `axis` and `keepdims` keyword only should be enough.